taskcluster/ci/nimbus-fml/kind.yml

```diff
@@ -0,0 +1,55 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# ⚠️ If you add, rename or delete a task here, please also update .mergify.yml! ⚠️
+---
+loader: taskgraph.loader.transform:loader
+
+transforms:
+  - app_services_taskgraph.transforms.appservices:transforms
+  - taskgraph.transforms.job:transforms
+  - taskgraph.transforms.task:transforms
+
+kind-dependencies:
+  - toolchain
+
+tasks:
+  build:
+    attributes:
+      run-on-pr-type: all
+      resource-monitor: true
+    needs-sccache: false # TODO: Bug 1623426 deal with this once we're in prod
+    run-on-tasks-for: [github-pull-request, github-release, cron]
+    description: Build and test (Swift)
+    scopes:
+      - project:releng:services/tooltool/api/download/internal
+    worker-type: b-linux
+    worker:
+      docker-image: { in-tree: linux }
+      max-run-time: 1800
+      artifacts:
+        - name: "public/build/nimbus-fml.zip"
+          path: "/builds/worker/checkouts/vcs/build/nimbus-fml.zip"
+          type: "file"
+        - name: "public/build/nimbus-fml.sha256"
+          path: "/builds/worker/checkouts/vcs/build/nimbus-fml.sha256"
+          type: "file"
+    run:
+      pre-commands:
+        - [git, submodule, update, --init]
+        - [source, taskcluster/scripts/toolchain/setup-fetched-rust-toolchain.sh]
+        - [rsync, '-a', /builds/worker/fetches/libs/, /builds/worker/checkouts/vcs/libs/]
+      commands:
+        - [
+            "/builds/worker/checkouts/vcs/taskcluster/scripts/build-nimbus-fml.py",
+            "/builds/worker/checkouts/vcs/build/",
+          ]
+      using: run-commands
+      use-caches: true
+    fetches:
+      toolchain:
+        - desktop-linux-libs
+        - rust
+    routes:
+      - index.project.application-services.v2.nimbus-fml.{appservices_version}
```

Comment:
This commit moves some of our artifact storage from github releases to taskcluster.  How long are indexed task artifacts stored?
  - I'm hoping that the nightly artifacts would go back at least 6 months
  - I'm hoping release artifacts would never expire.  Is that possible with taskcluster?  If not, I'm thinking we could do a github release for those.  Are there any other options?

@ahal @JohanLorenzo

Reply:
I think I've figured this out:
  - [All indexed entities must have an expiration](https://docs.taskcluster.net/docs/reference/core/index#entry-expiration).  I think this means we should do a github release for our non-nightly releases.
  - Our nightly releases are currently set to expire after 28 days.  I'll bump this to 180.

Reply:
Could you provide a link to the task please?

Artifact expiry and task expiry are distinct from one another, so one possibility is that you are looking at the artifact expiry locally. Another possibility is that the expiry value is being overwritten by some later phase of the generation (e.g a morph?), though I'm not aware of anything that should be doing this off the top of my head.

Reply:
This took me way too long to track down, but these are the Taskgraph defaults for try (or pull requests) vs pushes respectively:
https://github.com/taskcluster/taskgraph/blob/main/src/taskgraph/transforms/task.py#L1027

So I think everything is working as expected? You can add an `expires-after` key to the task definition if you'd like to have your own custom expiry / behaviour.

Reply:
Specifically the parameter file you used locally has `tasks_for` set to `cron`, and that `is_try()` function only returns True if it's set to `github-pull-request`:
https://github.com/taskcluster/taskgraph/blob/main/src/taskgraph/parameters.py#L226